src/vendor/webauthn.js

```diff
@@ -0,0 +1,185 @@
+            db = req.result;
+            resolve();
+          };
+          req.onerror = function (e) {
+            reject(e);
+          };
+        });
+      }
+
+      function store(id, data) {
+        if (!initPromise) {
+          initPromise = initDB();
+        }
+        return initPromise.then(function () {
+          doStore(id, data)
+        });
+      }
+
+      function doStore(id, data) {
+        if (!db) throw "DB not initialised";
+        return new Promise(function (resolve, reject) {
+          var tx = db.transaction(WEBAUTHN_ID_TABLE, "readwrite");
+          var store = tx.objectStore(WEBAUTHN_ID_TABLE);
+          store.put({id: id, data: data});
+          tx.oncomplete = function () {
+            resolve();
+          };
+          tx.onerror = function (e) {
+            reject(e);
+          };
+        });
+      }
+
+      function getAll() {
+        if (!initPromise) {
+          initPromise = initDB();
+        }
+        return initPromise.then(doGetAll);
+      }
+
+      function doGetAll() {
+        if (!db) throw "DB not initialised";
+        return new Promise(function (resolve, reject) {
+          var tx = db.transaction(WEBAUTHN_ID_TABLE, "readonly");
+          var store = tx.objectStore(WEBAUTHN_ID_TABLE);
+          var req = store.openCursor();
+          var res = [];
+          req.onsuccess = function () {
+            var cur = req.result;
+            if (cur) {
+              res.push({id: cur.value.id, data: cur.value.data});
+              cur.continue();
+            } else {
+              resolve(res);
+            }
+          };
+          req.onerror = function (e) {
+            reject(e);
+          };
+        });
+      }
+
+      return {
+        store: store,
+        getAll: getAll
+      };
+    }());
+
+    function makeCredential(accountInfo, cryptoParams, attestChallenge) {
+      var acct = {rpDisplayName: accountInfo.rpDisplayName, userDisplayName: accountInfo.displayName};
+      var params = [];
+      var i;
+
+      if (accountInfo.name) {
+        acct.accountName = accountInfo.name;
+      }
+      if (accountInfo.id) {
+        acct.userId = accountInfo.id;
+      }
+      if (accountInfo.imageUri) {
+        acct.accountImageUri = accountInfo.imageUri;
+      }
+
+      for (i = 0; i < cryptoParams.length; i++) {
+        if (cryptoParams[i].type === 'ScopedCred') {
+          params[i] = {type: 'FIDO_2_0', algorithm: cryptoParams[i].algorithm};
+        } else {
+          params[i] = cryptoParams[i];
+        }
+      }
+      return msCredentials.makeCredential(acct, params, attestChallenge).then(function (cred) {
+        if (cred.type === "FIDO_2_0") {
+          var result = Object.freeze({
+            credential: {type: "ScopedCred", id: cred.id},
+            publicKey: JSON.parse(cred.publicKey),
+            attestation: cred.attestation
+          });
+          return webauthnDB.store(result.credential.id, accountInfo).then(function () {
```

Comment:
we can remove this too. We don't need to store the credential id in the local storage as we store it on the server side.